src/ExcelIO/RowUpdate.cs

```diff
@@ -52,7 +52,7 @@ private static int UpdateRow
                 List<Cell> cells = rows[i].Elements<Cell>().ToList();
                 for (int j = 0; j < cells.Count; j++)
                 {
-                    Cell cell = cells[i];
+                    Cell cell = cells[j];
```

Comment:
This was the cause of the bug. I wrote 'i' (RowIndexer) instead of 'j' (CellIndexer).